src/main/java/executor/service/service/impl/ScenarioSourceListenerImpl.java

```diff
@@ -37,4 +33,22 @@ private void validateScenarios(List<Scenario> scenarios) {
             throw new IllegalArgumentException("List cannot be null or empty");
         }
     }
+
+    private void listen(List<Scenario> scenarioList, ScenarioHandler handler) {
+        int currentIndex = 0;
+        while(true){
```

Comment:
while' statement cannot complete without throwing an exception. This is not very good practice. 

In order to solve this problem, you need to redo the code following the example of ProxySourcesClient, pull-request [#43](https://github.com/GFLCourses5/worker/pull/43). 
This will allow in ParalleFlowExecutorService, pull-request [#30](https://github.com/GFLCourses5/worker/pull/30) to solve this problem correctly. 
Here, when calling the **public void shutdown()** method, the flag is switched correctly and the ExecutorService is completed.
![image](https://github.com/GFLCourses5/worker/assets/140727960/4a25fdc7-6b27-4f0f-bfb5-9b427e23b0d8)